### ISSUE TITLE
License Decision!

### ISSUE THREAD

**Issue by benkolera** [Dulux-Oz/DeadlandsClassic#1]
Add your ideas here as comments and we can vote.

**Comment by benkolera** [comment]
MIT

**Comment by haxxxton** [comment]
> The MIT license is permissive, placing very few restrictions on how the software can be used, while the GPL license is restrictive, requiring that any changes made to the code be released under the same license, and any software that uses the code must also be released under the same license.
Source [link](https://www.endorlabs.com/blog/open-source-licensing-simplified-a-comparative-overview-of-popular-licenses#:~:text=The%20MIT%20license%20is%20permissive,released%20under%20the%20same%20license.)


GPL for me, but not a deal breaker

**Comment by Ampersandrew** [comment]
I also prefer GPL, but also not rabidly.

**Comment by Dulux-Oz** [comment]
I'd also prefer to use the GPL (v3) but like everyone else if people have different opinions, etc...

**Comment by TheSpyder** [comment]
That's 3 against 2, GPL wins?

**Comment by Dulux-Oz** [comment]
Yeah, but the "Win" is a *preference*, so let's see what develops, shall we?

**Comment by Dulux-Oz** [comment]
OK, so as there's been no further comment/movement on this, I'll assume the current consensus of using the GPLv3 is what we are going with. In such a case, I'll close this off and make the necessary additions to the repo, etc (as a feature branch for both dev and main).

**Comment by Dulux-Oz** [comment]
So, after reading through all the relevant details about using the GPLv3, we need to add a Copyright Statement along the lines of:

- `Copyright (c) 2023 <Andrew>, <Andy>, <Ben>, Matthew J Black, <Haxxx>  & <James>`

So if you'd like to let me know your "preferred" (ie real) names to add to the above line, I'll make the necessary changes. I'll also list us all in (last name) alphabetical order - so there's no impression of favoritism, seniority, etc.

On a related tangent: When you get around to doing a commit could you each please finish filling out the relevant "authors" details of the system.json file, or remove the un-required tags (or I can do it for you if you let me know the details). Also, let's rearrange those sub-objects alphabetical by last name (or only name) as well - again, so there's no impression of favoritism, etc

**Comment by Dulux-Oz** [comment]
Also, there should be a "copyright block" at the start of each file. I'll add the block to each file as part of the feature/license_update branch I'm working on, but in the meantime you can (can you please) add the following block to each relevant file - cheers.

HTML Example:
~~~
<!--
################################################################################
## Copyright (c) 2023 {{Andrew}}, {{Andy}}, {{Ben}}, Matthew J Black,         ##
##    {{Haxxx}}  & {{James}}.                                                 ##
## This file is part of the *Deadlands Classic* Foundry Virtual Tabletop Game ##
##   System.                                                                  ##
## *Deadlands Classic* is free software: you can redistribute it and/or       ##
##    modify it under the terms of the GNU General Public License as          ##
##    published by the Free Software Foundation, either version 3 of the      ##
##    License, or (at your option) any later version.                         ##
## *Deadlands Classic* is distributed in the hope that it will be useful, but ##
##    WITHOUT ANY WARRANTY; without even the implied warranty of              ##
##    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU        ##
##    General Public License for more details.                                ##
## You should have received a copy of the GNU General Public License along    ##
##    with *Deadlands Classic*. If not, see <https://www.gnu.org/licenses/>.  ##
################################################################################
-->
~~~

Obviously for the JS files substiture `/*` and `*/` for `<!--` and `-->`, respectively.  :-)

Edit: Forgot the copyright part :-)

**Comment by Dulux-Oz** [comment]
Sorry, thought I'd already re-opened this - DOH!

**Comment by TheSpyder** [comment]
> Also, there should be a "copyright block" at the start of each file

This is not required - the repository license covers every file. Some projects do it to remind the casual observer that the file is under a specific license, but I've always found this tendency annoying 🙂

**Comment by Dulux-Oz** [comment]
OK, so reading the GPL doco they say to do it, but I'm happy to go with the majority :-)

**Comment by TheSpyder** [comment]
I did some research into this for TinyMCE a year or two ago. The recommendation is there as a backup, and because so many people do it, but legally it makes no difference.

It's a bit like "all rights reserved" that appears in some licenses - that hasn't been required since a worldwide copyright agreement in the 80s, but the habit persists.

**Comment by Dulux-Oz** [comment]
So, what's the consensus here - add it to each file in dev or not, because FVTT does use "compiled" files (unless we want it too).

I'm confused...  :-(

**Comment by haxxxton** [comment]
my personal preference is to not put it on every file. its likely we end up moving towards a working files directory, and an output directory. in that case we could automate the addition of the header to the output files, but its just going to get in the way of working files IMO

**Comment by Dulux-Oz** [comment]
I can work with that. What about everyone else?

**Comment by TheSpyder** [comment]
It's basically personal choice. And I definitely fall on the side of not on every file - when TinyMCE switched to MIT I deleted all the file LGPL headers :)
https://github.com/tinymce/tinymce/pull/7647

**Comment by Dulux-Oz** [comment]
Alright, makes sense.

Let's wait a day or two for Ben and Andrew (& James) to weigh in before closing this off and moving ahead, OK?

Edit: speelling is hard

**Comment by haxxxton** [comment]
Solved in #5
